spec/models/task_spec.rb

```diff
@@ -12,20 +12,40 @@
     expect(Task.new.status).to eq 'unstarted'
   end
 
-  describe '#latest_submission' do
+  describe '#active_submission' do
     let(:task) { FactoryBot.create(:task) }
+    let(:task2) { FactoryBot.create(:task) }
+    let!(:unrelated_submission) { FactoryBot.create(:submission, task: task2, aasm_state: 'completed') }
 
-    it 'returns the most recent submission' do
-      _old_submission = FactoryBot.create(:submission, task: task)
+    context 'there is no completed submission' do
+      let!(:old_submission) { FactoryBot.create(:submission, task: task, created_at: 2.days.ago) }
+      let!(:new_submission) { FactoryBot.create(:submission, task: task, created_at: 1.day.ago) }
 
-      travel 1.day do
-        latest_submission = FactoryBot.create(:submission, task: task)
+      it 'returns the most recent submission' do
+        expect(task.active_submission).to eq(new_submission)
+      end
+    end
+
+    context 'there is a completed submission that is not the most recent' do
+      let!(:old_submission) { FactoryBot.create(:submission, task: task, created_at: 3.days.ago) }
```

Comment:
Is this submission's `create` necessary?

Reply:
If we don't create it, then we don't have a old submission in the database that we want to check isn't returned by #active_submission

Reply:
reasonable. It is a tired late Thu question 😪